src/basic/in-addr-util.c

```diff
@@ -464,3 +464,45 @@ int in_addr_mask(int family, union in_addr_union *addr, unsigned char prefixlen)
 
         return -EAFNOSUPPORT;
 }
+
+int in_addr_prefix_from_string(const char *p, int family, union in_addr_union *prefix, uint8_t *prefixlen) {
+        union in_addr_union buffer;
+        const char *e, *l;
+        uint8_t k;
+        int f, r;
+
+        assert(p);
+
+        e = strchr(p, '/');
+        if (e)
+                l = strndupa(p, e - p);
+        else
+                l = p;
+
+        r = in_addr_from_string_auto(l, &f, &buffer);
+        if (r < 0)
+                return r;
+
+        if (f != family)
+                return -1;
+
+        k = FAMILY_ADDRESS_SIZE(f) * 8;
+
+        if (e) {
+                uint8_t n;
+
+                r = safe_atou8(++e, &n);
```

Comment:
why use "e++"? this suggests you actually care about e after this call, and that it is once char further. But you actually don't. Hence this should really just read "e+1"